src/AuthorizedSocketConnection.js

```diff
@@ -81,19 +81,24 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
     this.socket.emit('app_error', error);
   }
 
-  isAuthorized = (data: any) => {
+  isAuthorized = (
+    hasPermission: (data: any, credentials: TCredentials) => boolean,
+  ) => {
     const credentials = this.config.credentialsManager.getCredentials();
-    const isAuthorized =
-      !!credentials && this.config.hasPermission(data, credentials);
-
-    if (!isAuthorized) {
-      this.log('info', 'unauthorized', {
-        payload: data,
-        credentials,
-      });
-    }
+    return (data: any) => {
```

Comment:
@taion the argument we pass into `AsyncUtils.filter` still has to be a function that takes `data` as input

Reply:
I think the idea is to inline the filter function outside of `isAuthorized` where you use it.

```js
AsyncUtils.filter(
  await subscriptionContext.subscribe(topic, options),
  data => this.isAuthorized(data, hasPermission),
)
```

Reply:
ah! ok